sos/report/plugins/drbd.py

```diff
@@ -7,27 +7,74 @@
 # See the LICENSE file in the source distribution for further information.
 
 from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+from sos.utilities import find
 
 
-class DRDB(Plugin, RedHatPlugin, UbuntuPlugin):
+class DRBD(Plugin, RedHatPlugin, UbuntuPlugin):
 
     short_desc = 'Distributed Replicated Block Device (DRBD)'
 
     plugin_name = 'drbd'
     profiles = ('storage',)
-    packages = ('drbd*-utils',)
+    commands = ('drbdsetup',)
+
+    # In case of kernel bugs, drbdsetup may hang indefinitely.
+    # Set a shorter than default timeout.
+    cmd_timeout = 60
+
+    def add_drbd_thread_stacks(self):
+        stacks = ""
+        for pid_file in find("*_pid", "/sys/kernel/debug/drbd/resources/"):
+            with open(pid_file, 'r', encoding='utf-8') as f:
+                pid = f.read().strip()
+            stacks += f"--- {pid_file}: {pid} ---\n"
+            if pid.isdigit():
+                try:
+                    sfn = f"/proc/{pid}/stack"
+                    with open(sfn, 'r', encoding='utf-8') as sf:
+                        stacks += sf.read()
+                except Exception as e:
+                    stacks += f"Could not read /proc/{pid}/stack: {e}\n"
+            stacks += "\n"
+
+        self.add_string_as_file(stacks, "drbd_thread_stacks")
 
     def setup(self):
         self.add_cmd_output([
-            "drbd-overview",
-            "drbdadm dump-xml",
-            "drbdsetup status all",
+            "drbdadm dump",
+            "drbdadm -d -vvv adjust all",
+            "drbdsetup status -vs all",
             "drbdsetup show all"
         ])
+        self.add_drbd_thread_stacks()
+        for kmod in find("drbd*.ko*", "/lib/modules"):
+            self.add_cmd_output([
+                f"modinfo {kmod}",
+            ], suggest_filename=f"modinfo_{kmod.replace('/', '_')}")
         self.add_copy_spec([
             "/etc/drbd.conf",
             "/etc/drbd.d/*",
-            "/proc/drbd"
+            "/proc/drbd",
+            "/sys/kernel/debug/drbd/*",
+            "/var/lib/drbd/*",
+            "/var/lib/drbd-support/*",
+            "/var/lib/linstor.d/*"
         ])
 
+    def postproc(self):
+        # Scrub nodehash from /var/lib/drbd-support/registration.json
+        nodehash_re = r'("nodehash"\s*:\s*")[a-zA-Z0-9]+"'
```

Comment:
You *might* shorten the `[a-zA-Z0-9]` by simple and elegant `\w` but that will catch also `_`  you might not want to be matched. So / anyway ack with the current RE.